src/faq/troubleshooting.rst

```diff
@@ -0,0 +1,222 @@
+you effectively copy your entire data into the view index. This takes a lot of
+time. Consider rewriting your emit() call to emit(key, null); and query the view
+with the `?include_docs=true parameter` to get all doc's data with the view
+without having to copy the data into the view index.
+
+How can I "JOIN" with CouchDB?
+------------------------------
+
+See https://www.cmlenz.net/archives/2007/10/couchdb-joins
+
+Why is my database so large, even after compaction?
+---------------------------------------------------
+
+Often, CouchDB users expect that adding a document to a database, then deleting
+that document will return the database to its original state. However, this is
+not the case. Consider a two-database scenario:
+
+- Doc 1 inserted to DB A.
+- DB A replicated to DB B.
+- Doc 1 deleted from DB A.
+- DB A replicated to DB B.
+
+If inserting and then deleting a document returned the database to the original
+state, the second replication from A to B would be "empty" and hence DB B would
+be unchanged, which means it would be out of sync with DB A.
+
+To handle this case, CouchDB keeps a record of each document deleted, by keeping
+the document _id, _rev and _deleted=true. The data size per deleted doc depends
+on the number of revisions that CouchDB has to track plus the data size for any
+data stored in the deleted revision (this is usually relatively small, kilobytes
+perhaps, but varies based on use case). It is possible to keep audit trail data
+with a deleted document (ie. application-specific things like "deleted_by" and
+"deleted_at"). While generally this is not an issue, if the DB is still larger
+than expected, even after considering the minimum size of a deleted document,
+check to insure that the deleted document doesn't contain data not unintended
+for keeping past the deletion action. Specifically, if your client library is
+not careful, it could be storing a full copy of each document in the deleted
+revisions. For more information:
+https://issues.apache.org/jira/browse/COUCHDB-1141
+
+My database will require an unbounded number of deletes, what can I do?
+-----------------------------------------------------------------------
+
+If there's a strong correlation between time (or some other regular
+monotonically increasing event) and document deletion, a DB setup can be used
+like the following:
+
+- Assume that the past 30 days of logs are needed, anything older can be
+  deleted.
+- Set up DB logs_2011_08.
+- Replicate logs_2011_08 to logs_2011_09, filtered on logs from 2011_08 only.
+- During August, read/write to logs_2011_08.
+- When September starts, create logs_2011_10.
+- Replicate logs_2011_09 to logs_2011_10, filtered on logs from 2011_09 only.
+- During September, read/write to logs_2011_09.
+- Logs from August will be present in logs_2011_09 due to the replication, but
+  not in logs_2011_10.
+- The entire logs_2011_08 DB can be removed.
```

Comment:
This should include the entire approach described by @willholley in the comments of this blog post from 2015:

https://eclipsesource.com/blogs/2015/04/20/how-to-finally-delete-documents-in-couchdb/

I'm sure he wouldn't mind us cribbing his text for here. :)